### ISSUE TITLE
Deployment on Bielefeld OpenStack failled

### ISSUE THREAD

**Issue by sneumann** [kubenow/KubeNow#33]
Hi,
Issue to lift a kubernetes cluster. Tried with terraform 0.7.8 and 0.7.9. 

Yours, Steffen

```
root@host-192-168-0-5:~/KubeNow# terraform apply openstack # deploy the cluster
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Warnings:

  * template_file.master_bootstrap: using template_file as a resource is deprecated; consider using the data source instead
  * template_file.edge_bootstrap: using template_file as a resource is deprecated; consider using the data source instead
  * template_file.node_bootstrap: using template_file as a resource is deprecated; consider using the data source instead

No errors found. Continuing with 3 warning(s).

module.master.template_file.master_bootstrap: Creating...
  rendered:           "" => "<computed>"
  template:           "" => "#!/bin/bash\n\n# Exit immediately if a command exits with a non-zero status\nset -e\n\necho \"Inititializing the master...\"\nkubeadm init --token ${kubeadm_token}\n"
  vars.%:             "" => "1"
  vars.kubeadm_token: "" => "3cc7...14f8"
module.master.template_file.master_bootstrap: Creation complete
module.edge.openstack_compute_floatingip_v2.edge_ip.0: Creating...
  address:     "" => "<computed>"
  fixed_ip:    "" => "<computed>"
  instance_id: "" => "<computed>"
  pool:        "" => "cebitec"
  region:      "" => "RegionOne"
module.edge.openstack_compute_floatingip_v2.edge_ip.1: Creating...
  address:     "" => "<computed>"
  fixed_ip:    "" => "<computed>"
  instance_id: "" => "<computed>"
  pool:        "" => "cebitec"
  region:      "" => "RegionOne"
module.network.openstack_compute_secgroup_v2.main: Creating...
  description:                   "" => "The automatically created secgroup for KubeNow"
  name:                          "" => "KubeNow-secgroup"
  region:                        "" => "RegionOne"
  rule.#:                        "" => "5"
  rule.112275015.cidr:           "" => ""
  rule.112275015.from_group_id:  "" => ""
  rule.112275015.from_port:      "" => "1"
  rule.112275015.id:             "" => "<computed>"
  rule.112275015.ip_protocol:    "" => "tcp"
  rule.112275015.self:           "" => "true"
  rule.112275015.to_port:        "" => "65535"
  rule.3211544828.cidr:          "" => "0.0.0.0/0"
  rule.3211544828.from_group_id: "" => ""
  rule.3211544828.from_port:     "" => "443"
  rule.3211544828.id:            "" => "<computed>"
  rule.3211544828.ip_protocol:   "" => "tcp"
  rule.3211544828.self:          "" => "false"
  rule.3211544828.to_port:       "" => "443"
  rule.3719211069.cidr:          "" => "0.0.0.0/0"
  rule.3719211069.from_group_id: "" => ""
  rule.3719211069.from_port:     "" => "80"
  rule.3719211069.id:            "" => "<computed>"
  rule.3719211069.ip_protocol:   "" => "tcp"
  rule.3719211069.self:          "" => "false"
  rule.3719211069.to_port:       "" => "80"
  rule.635693822.cidr:           "" => ""
  rule.635693822.from_group_id:  "" => ""
  rule.635693822.from_port:      "" => "1"
  rule.635693822.id:             "" => "<computed>"
  rule.635693822.ip_protocol:    "" => "udp"
  rule.635693822.self:           "" => "true"
  rule.635693822.to_port:        "" => "65535"
  rule.836640770.cidr:           "" => "0.0.0.0/0"
  rule.836640770.from_group_id:  "" => ""
  rule.836640770.from_port:      "" => "22"
  rule.836640770.id:             "" => "<computed>"
  rule.836640770.ip_protocol:    "" => "tcp"
  rule.836640770.self:           "" => "false"
  rule.836640770.to_port:        "" => "22"
module.network.openstack_networking_router_v2.main: Creating...
  admin_state_up:   "" => "<computed>"
  distributed:      "" => "<computed>"
  external_gateway: "" => "9e...a95db6e3"
  name:             "" => "KubeNow-router"
  region:           "" => "RegionOne"
  tenant_id:        "" => "<computed>"
module.network.openstack_networking_network_v2.main: Creating...
  admin_state_up: "" => "true"
  name:           "" => "KubeNow-network"
  region:         "" => "RegionOne"
  shared:         "" => "<computed>"
  tenant_id:      "" => "<computed>"
module.master.openstack_compute_floatingip_v2.master_ip: Creating...
  address:     "" => "<computed>"
  fixed_ip:    "" => "<computed>"
  instance_id: "" => "<computed>"
  pool:        "" => "cebitec"
  region:      "" => "RegionOne"
module.keypair.openstack_compute_keypair_v2.main: Creating...
  name:       "" => "KubeNow-keypair"
  public_key: "" => "ssh-rsa AA.......\n\n\x1b[?2004l"
  region:     "" => "RegionOne"
module.edge.openstack_compute_floatingip_v2.edge_ip.0: Creation complete
module.keypair.openstack_compute_keypair_v2.main: Creation complete
module.master.openstack_compute_floatingip_v2.master_ip: Creation complete
module.edge.openstack_compute_floatingip_v2.edge_ip.1: Creation complete
module.network.openstack_compute_secgroup_v2.main: Creation complete
module.network.openstack_networking_router_v2.main: Creation complete
module.network.openstack_networking_network_v2.main: Creation complete
module.master.openstack_compute_instance_v2.master: Creating...
  access_ip_v4:               "" => "<computed>"
  access_ip_v6:               "" => "<computed>"
  flavor_id:                  "" => "<computed>"
  flavor_name:                "" => "c2r30d160"
  floating_ip:                "" => "172.21.8.159"
  image_id:                   "" => "<computed>"
  image_name:                 "" => "KubeNow"
  key_pair:                   "" => "KubeNow-keypair"
  name:                       "" => "KubeNow-master"
  network.#:                  "" => "1"
  network.0.access_network:   "" => "false"
  network.0.fixed_ip_v4:      "" => "<computed>"
  network.0.fixed_ip_v6:      "" => "<computed>"
  network.0.floating_ip:      "" => "<computed>"
  network.0.mac:              "" => "<computed>"
  network.0.name:             "" => "KubeNow-network"
  network.0.port:             "" => "<computed>"
  network.0.uuid:             "" => "<computed>"
  region:                     "" => "RegionOne"
  security_groups.#:          "" => "1"
  security_groups.1881937636: "" => "KubeNow-secgroup"
  stop_before_destroy:        "" => "false"
  user_data:                  "" => "a399...a5bf"
module.network.openstack_networking_subnet_v2.main: Creating...
  cidr:                       "" => "10.0.0.0/16"
  dns_nameservers.#:          "" => "2"
  dns_nameservers.3811298194: "" => "8.8.4.4"
  dns_nameservers.3817307869: "" => "8.8.8.8"
  enable_dhcp:                "" => "true"
  gateway_ip:                 "" => "<computed>"
  ip_version:                 "" => "4"
  name:                       "" => "KubeNow-subnet"
  network_id:                 "" => "99...aac80"
  region:                     "" => "RegionOne"
  tenant_id:                  "" => "<computed>"
module.network.openstack_networking_subnet_v2.main: Creation complete
module.network.openstack_networking_router_interface_v2.main: Creating...
  region:    "" => "RegionOne"
  router_id: "" => "8cd6..d45f043"
  subnet_id: "" => "5b5a...b8612"
module.network.openstack_networking_router_interface_v2.main: Creation complete
Error applying plan:

1 error(s) occurred:

* openstack_compute_instance_v2.master: json: cannot unmarshal number 1.994752e+06 into Go value of type int

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

OpenStack API version 2.0

**Comment by mcapuccini** [comment]
As reported in https://github.com/hashicorp/terraform/issues/10130, a workaround to this issue consists in using Terraform 0.7.7.

**Comment by mcapuccini** [comment]
Setting `wontfix` as it is a problem caused by https://github.com/gophercloud/gophercloud. Hopefully it is going to be solved soonish.

**Comment by andersla** [comment]
A workaround is to specify flavor_id instead of flavor_name, or image_id instead of image_name, also reported in hashicorp/terraform#10130

**Comment by mcapuccini** [comment]
@sneumann is this still a problem?

**Comment by andersla** [comment]
Fixed by Terraform upgrade to version > 0.9.4

**Comment by mcapuccini** [comment]
closing then 🙂
